core/src/main/java/org/apache/spark/memory/MemoryConsumer.java

```diff
@@ -154,8 +156,10 @@ private void throwOom(final MemoryBlock page, final long required) {
     }
     taskMemoryManager.showMemoryUsage();
     // checkstyle.off: RegexpSinglelineJava
-    throw new SparkOutOfMemoryError("UNABLE_TO_ACQUIRE_MEMORY",
-            new String[]{Long.toString(required), Long.toString(got)});
+    Map<String, String> params = new HashMap<String, String>();
+    params.put("requestedBytes", Long.toString(required));
+    params.put("receivedBytes", Long.toString(got));
+    throw new SparkOutOfMemoryError("UNABLE_TO_ACQUIRE_MEMORY", params);
```

Comment:
shall we move it to `SparkCoreErrors`? Then we don't need to create a map in Java.